js/engine_unit_state.js

```diff
@@ -0,0 +1,448 @@
+    const newEquipment = unitConverter.convert(newEquipmentRaw, "units").getValue();
+
+    const deltaValue = newEquipment - priorEquipment;
+    return new EngineNumber(deltaValue, "units");
+  }
+}
+
+/**
+ * State getter that allows overriding values from an inner state getter.
+ */
+class OverridingConverterStateGetter {
+  /**
+   * Create a new overriding converter state getter.
+   *
+   * @param {ConverterStateGetter} innerGetter - The base state getter to wrap.
+   */
+  constructor(innerGetter) {
+    const self = this;
+    self._innerGetter = innerGetter;
+    self._energyIntensity = null;
+    self._amortizedUnitVolume = null;
+    self._population = null;
+    self._yearsElapsed = null;
+    self._totalConsumption = null;
+    self._energyConsumption = null;
+    self._volume = null;
+    self._amortizedUnitConsumption = null;
+    self._populationChange = null;
+    self._substanceConsumption = null;
+  }
+
+  /**
+   * Set total values for different stream types.
+   *
+   * @param {string} streamName - The name of the stream (sales, manufacture,
+   *     import, etc.).
+   * @param {EngineNumber} value - The value to set for the stream.
+   */
+  setTotal(streamName, value) {
+    const self = this;
+    const strategy = {
+      sales: (x) => self.setVolume(x),
+      manufacture: (x) => self.setVolume(x),
+      import: (x) => self.setVolume(x),
+      equipment: (x) => self.setPopulation(x),
+      priorEquipment: (x) => self.setPopulation(x),
+      consumption: (x) => self.setConsumption(x),
+    }[streamName];
+    strategy(value);
+  }
+
+  /**
+   * Set the substance consumption value.
+   *
+   * @param {EngineNumber} newValue - The new substance consumption value.
+   */
+  setSubstanceConsumption(newValue) {
+    const self = this;
+    self._substanceConsumption = newValue;
+  }
+
+  /**
+   * Get the substance consumption value.
+   *
+   * @returns {EngineNumber} The substance consumption value.
+   */
+  getSubstanceConsumption() {
+    const self = this;
+    if (self._substanceConsumption === null) {
+      return self._innerGetter.getSubstanceConsumption();
+    } else {
+      return self._substanceConsumption;
+    }
+  }
+
+  /**
+   * Set the energy intensity.
+   *
+   * @param {EngineNumber} newValue - The new energy intensity value.
+   */
+  setEnergyIntensity(newValue) {
+    const self = this;
+    self._energyIntensity = newValue;
+  }
+
+  /**
+   * Get the energy intensity.
+   *
+   * @returns {EngineNumber} The energy intensity value with units.
+   */
+  getEnergyIntensity() {
+    const self = this;
+    if (self._energyIntensity === null) {
+      return self._innerGetter.getEnergyIntensity();
+    } else {
+      return self._energyIntensity;
+    }
+  }
+
+  /**
+   * Set the amortized unit volume.
+   *
+   * @param {EngineNumber} newValue - The new amortized unit volume.
+   */
+  setAmortizedUnitVolume(newValue) {
```

Comment:
Can you please update the JSDoc in engine_unit_state to clarify that AmortizedUnitVolume Refers to the amortized initial charge specifically?